src/node_serdes.cc

```diff
@@ -0,0 +1,457 @@
+}
+
+void DeserializerContext::ReadRawBytes(
+    const FunctionCallbackInfo<Value>& args) {
+  DeserializerContext* ctx;
+  ASSIGN_OR_RETURN_UNWRAP(&ctx, args.Holder());
+
+  const void* data;
+  size_t length = args[0]->NumberValue();
+  bool ok = ctx->deserializer_.ReadRawBytes(length, &data);
+  if (!ok) return ctx->env()->ThrowError("ReadRawBytes() failed");
+
+  const uint8_t* position = reinterpret_cast<const uint8_t*>(data);
+  CHECK_GE(position, ctx->data_);
+  CHECK_LE(position + length, ctx->data_ + ctx->length_);
+
+  const uint32_t offset = position - ctx->data_;
+  CHECK_EQ(ctx->data_ + offset, position);
+
+  args.GetReturnValue().Set(offset);
+}
+
+void InitializeSerdesBindings(Local<Object> target,
+                              Local<Value> unused,
+                              Local<Context> context) {
+  Environment* env = Environment::GetCurrent(context);
+  Local<FunctionTemplate> ser =
+      env->NewFunctionTemplate(SerializerContext::New);
+
+  ser->InstanceTemplate()->SetInternalFieldCount(1);
+
+  env->SetProtoMethod(ser, "writeHeader", SerializerContext::WriteHeader);
+  env->SetProtoMethod(ser, "writeValue", SerializerContext::WriteValue);
+  env->SetProtoMethod(ser, "releaseBuffer", SerializerContext::ReleaseBuffer);
+  env->SetProtoMethod(ser,
+                      "transferArrayBuffer",
+                      SerializerContext::TransferArrayBuffer);
+  env->SetProtoMethod(ser, "writeUint32", SerializerContext::WriteUint32);
+  env->SetProtoMethod(ser, "writeUint64", SerializerContext::WriteUint64);
+  env->SetProtoMethod(ser, "writeDouble", SerializerContext::WriteDouble);
+  env->SetProtoMethod(ser, "writeRawBytes", SerializerContext::WriteRawBytes);
+
+  ser->SetClassName(FIXED_ONE_BYTE_STRING(env->isolate(), "Serializer"));
+  target->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "Serializer"),
+              ser->GetFunction());
+
+  Local<FunctionTemplate> des =
+      env->NewFunctionTemplate(DeserializerContext::New);
+
+  des->InstanceTemplate()->SetInternalFieldCount(1);
+
+  env->SetProtoMethod(des, "readHeader", DeserializerContext::ReadHeader);
+  env->SetProtoMethod(des, "readValue", DeserializerContext::ReadValue);
+  env->SetProtoMethod(des,
+                      "getWireFormatVersion",
+                      DeserializerContext::GetWireFormatVersion);
+  env->SetProtoMethod(des,
+                      "transferArrayBuffer",
+                      DeserializerContext::TransferArrayBuffer);
+  env->SetProtoMethod(des, "readUint32", DeserializerContext::ReadUint32);
+  env->SetProtoMethod(des, "readUint64", DeserializerContext::ReadUint64);
+  env->SetProtoMethod(des, "readDouble", DeserializerContext::ReadDouble);
+  env->SetProtoMethod(des, "_readRawBytes", DeserializerContext::ReadRawBytes);
+
+  des->SetClassName(FIXED_ONE_BYTE_STRING(env->isolate(), "Deserializer"));
+  target->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "Deserializer"),
+              des->GetFunction());
```

Comment:
Ditto.  There are a few other places where you use old-style (implicit-style?) methods, such as the no-arg overloads of NumberValue() and Uint32Value().